### PR TITLE
opencv: disable python detection + no autodetect of ccache

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/9983

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

File: recipes/opencv/2.x/conanfile.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def _configure_cmake(self):
         self._cmake.definitions["OPENCV_MODULES_PUBLIC"] = "opencv"
         self._cmake.definitions["BUILD_opencv_nonfree"] = self.options.nonfree
 
+        self._cmake.definitions["ENABLE_CCACHE"] = False
```
Comment (SSE4) [inline recipes/opencv/2.x/conanfile.py:196]: why no `OPENCV_PYTHON_SKIP_DETECTION` for 2.x?

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Visual Studio agents are not very reliables.

File: recipes/opencv/2.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/opencv/2.x/conanfile.py:201]: disabled in opencv 2.x by https://github.com/conan-io/conan-center-index/pull/9995/commits/134884423ed3647837f7366357dfcacce90ff959

**Comment by SpaceIm** [comment]
@jgsogo all build succeed except opencv/2.4.13.7 static with Visual Studio 2019 & MD runtime:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003: The specified task executable "Lib.exe" could not be run. System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.dir\Release\opencv_f.CE358F0A.tlog'. [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at System.IO.FileSystemEnumerableIterator`1.CommonInit() [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at System.IO.Directory.GetFiles(String path, String searchPattern) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at Microsoft.Build.Utilities.TrackedDependencies.ExpandWildcards(ITaskItem[] expand) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at Microsoft.Build.Utilities.CanonicalTrackedOutputFiles.InternalConstruct(ITask ownerTask, ITaskItem[] tlogFiles, Boolean constructOutputsFromTLogs) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at Microsoft.Build.CPPTasks.TrackedVCToolTask.PostExecuteTool(Int32 exitCode) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at Microsoft.Build.CPPTasks.TrackedVCToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(1522,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.vcxproj]
```

What's wrong? There was no issue in previous PR.

**Comment by SSE4** [comment]
> C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.dir\Release\opencv_f.CE358F0A.tlog

may it be `MAX_PATH` limitation?
```
len(r"C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.dir\Release\opencv_f.CE358F0A.tlog")
== 218
```

**Comment by SpaceIm** [comment]
> > C:\J\w\prod\[BuildSingleReference@3.conan](mailto:BuildSingleReference@3.conan)\data\opencv\2.4.13.7__\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.dir\Release\opencv_f.CE358F0A.tlog
> 
> may it be `MAX_PATH` limitation?
> 
> ```
> len(r"C:\J\w\prod\BuildSingleReference@3\.conan\data\opencv\2.4.13.7\_\_\build\971796c24e4fb3a71aba8195a8fb841f8d181bae\build_subfolder\source_subfolder\modules\features2d\opencv_features2d.dir\Release\opencv_f.CE358F0A.tlog")
> == 218
> ```

I'l try to reduce path length, but why an error for this specific settings and not Visual Studio MDd or even same settings but shared flavor of opencv?

**Comment by SSE4** [comment]
> I'l try to reduce path length, but why an error for this specific settings and not Visual Studio MDd or even same settings but shared flavor of opencv?

to be honest, no idea. it would take a time to investigate. maybe instead we can just activate `LongPathsEnabled` in CI. it should help at least for some cases (ofcourse, it won't help if `msbuild.exe` contains a code like `char buffer[MAX_PATH]; GetModuleFileName(handle, buffer, MAX_PATH);`)

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]
wow, @SpaceIm, you got a green build!

**Comment by SpaceIm** [comment]
@ericLemanissier @prince-chrismc Could you review this PR please? Easy one ;)

**Review by ericLemanissier — APPROVED** [review]

**Comment by jgsogo** [comment]
> > I'l try to reduce path length, but why an error for this specific settings and not Visual Studio MDd or even same settings but shared flavor of opencv?
> 
> to be honest, no idea. it would take a time to investigate. maybe instead we can just activate `LongPathsEnabled` in CI. it should help at least for some cases (ofcourse, it won't help if `msbuild.exe` contains a code like `char buffer[MAX_PATH]; GetModuleFileName(handle, buffer, MAX_PATH);`)

Let's add `LongPathsEnabled` to the CI, now with the ansibles everything is easier (documented for the future us). Adding it to the backlog 👍

**Comment by jgsogo** [comment]
btw, merge for this recipe has failed:

```
ERROR: Missing binary: ffmpeg/4.4:16fe9749864b381504dff2f57f1b919aace58a77
ffmpeg/4.4: WARN: Can't find a 'ffmpeg/4.4' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Release, compiler=gcc, compiler.version=5, os=Linux
- Options: avcodec=True, avdevice=True, avfilter=True, avformat=True, fPIC=True, postproc=True, shared=False, swresample=True, swscale=True, with_bzip2=True, with_freetype=True, with_libalsa=True, with_libfdk_aac=True, with_libiconv=True, with_libmp3lame=True, with_libvpx=True, with_libwebp=True, with_libx264=True, with_libx265=True, with_lzma=True, with_openh264=True, with_openjpeg=True, with_opus=True, with_programs=True, with_pulse=True, with_sdl=False, with_ssl=openssl, with_vaapi=True, with_vdpau=True, with_vorbis=True, with_xcb=True, with_zeromq=False, with_zlib=True, brotli:fPIC=True, brotli:shared=False, bzip2:build_executable=True, bzip2:fPIC=True, bzip2:shared=False, flac:fPIC=True, flac:shared=False, freetype:fPIC=True, freetype:shared=False, freetype:subpixel=False, freetype:with_brotli=True, freetype:with_bzip2=True, freetype:with_png=True, freetype:with_zlib=True, libalsa:disable_python=True, libalsa:fPIC=True, libalsa:shared=False, libcap:fPIC=True, libcap:psx_syscals=False, libcap:shared=False, libfdk_aac:fPIC=True, libfdk_aac:shared=False, libiconv:fPIC=True, libiconv:shared=False, libmp3lame:fPIC=True, libmp3lame:shared=False, libpng:api_prefix=, libpng:fPIC=True, libpng:shared=False, libpng:sse=True, libsndfile:experimental=False, libsndfile:fPIC=True, libsndfile:programs=True, libsndfile:shared=False, libsndfile:with_alsa=False, libsndfile:with_external_libs=True, libtool:fPIC=True, libtool:shared=False, libvpx:avx=False, libvpx:avx2=False, libvpx:avx512=False, libvpx:fPIC=True, libvpx:mmx=True, libvpx:shared=False, libvpx:sse=True, libvpx:sse2=True, libvpx:sse3=True, libvpx:sse4_1=True, libvpx:ssse3=True, libwebp:fPIC=True, libwebp:near_lossless=True, libwebp:shared=False, libwebp:swap_16bit_csp=False, libwebp:with_simd=True, libx264:bit_depth=all, libx264:fPIC=True, libx264:shared=False, libx265:HDR10=False, libx265:SVG_HEVC_encoder=False, libx265:assembly=True, libx265:bit_depth=8, libx265:fPIC=True, libx265:shared=False, libx265:with_numa=False, ogg:fPIC=True, ogg:shared=False, openh264:fPIC=True, openh264:shared=False, openjpeg:build_codec=False, openjpeg:fPIC=True, openjpeg:shared=False, openssl:386=False, openssl:enable_weak_ssl_ciphers=False, openssl:fPIC=True, openssl:no_aria=False, openssl:no_asm=False, openssl:no_async=False, openssl:no_bf=False, openssl:no_blake2=False, openssl:no_camellia=False, openssl:no_cast=False, openssl:no_chacha=False, openssl:no_cms=False, openssl:no_comp=False, openssl:no_ct=False, openssl:no_deprecated=False, openssl:no_des=False, openssl:no_dgram=False, openssl:no_dh=False, openssl:no_dsa=False, openssl:no_dso=False, openssl:no_ec=False, openssl:no_ecdh=False, openssl:no_ecdsa=False, openssl:no_engine=False, openssl:no_filenames=False, openssl:no_gost=False, openssl:no_hmac=False, openssl:no_idea=False, openssl:no_md4=False, openssl:no_md5=False, openssl:no_mdc2=False, openssl:no_ocsp=False, openssl:no_pinshared=False, openssl:no_rc2=False, openssl:no_rfc3779=False, openssl:no_rmd160=False, openssl:no_rsa=False, openssl:no_seed=False, openssl:no_sha=False, openssl:no_sm2=False, openssl:no_sm3=False, openssl:no_sm4=False, openssl:no_sock=False, openssl:no_srp=False, openssl:no_srtp=False, openssl:no_sse2=False, openssl:no_ssl=False, openssl:no_ssl3=False, openssl:no_stdio=False, openssl:no_tests=False, openssl:no_threads=False, openssl:no_tls1=False, openssl:no_ts=False, openssl:no_whirlpool=False, openssl:openssldir=None, openssl:shared=False, opus:fPIC=True, opus:fixed_point=False, opus:shared=False, pulseaudio:fPIC=True, pulseaudio:shared=False, pulseaudio:with_alsa=True, pulseaudio:with_dbus=False, pulseaudio:with_glib=False, pulseaudio:with_openssl=True, pulseaudio:with_x11=True, vorbis:fPIC=True, vorbis:shared=False, xz_utils:fPIC=True, xz_utils:shared=False, zlib:fPIC=True, zlib:shared=False
- Dependencies: zlib/1.2.12, bzip2/1.0.8, xz_utils/5.2.5, libiconv/1.16, freetype/2.11.1, openjpeg/2.4.0, openh264/2.1.1, vorbis/1.3.7, opus/1.3.1, libx264/20191217, libx265/3.4, libvpx/1.11.0, libmp3lame/3.100, libfdk_aac/2.0.2, libwebp/1.2.2, openssl/1.1.1n, libalsa/1.2.5.1, xorg/system, pulseaudio/14.2, vaapi/system, vdpau/system
- Requirements: bzip2/1.Y.Z, freetype/2.Y.Z, libalsa/1.Y.Z, libfdk_aac/2.Y.Z, libiconv/1.Y.Z, libmp3lame/3.Y.Z, libvpx/1.Y.Z, libwebp/1.Y.Z, libx264/20191217.Y.Z, libx265/3.Y.Z, openh264/2.Y.Z, openjpeg/2.Y.Z, openssl/1.Y.Z, opus/1.Y.Z, pulseaudio/14.Y.Z, vaapi/system, vdpau/system, vorbis/1.Y.Z, xorg/system, xz_utils/5.Y.Z, zlib/1.Y.Z
- Package ID: 16fe9749864b381504dff2f57f1b919aace58a77
```

But that package of `ffmpeg/4.4` is available (with a different packageID):

https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/prod/TapaholesList/35/summary.json

There is some override there that is modifying that packageID

**Comment by SpaceIm** [comment]
I've found why. A dependency has been added to `pulseaudio` recipe in https://github.com/conan-io/conan-center-index/pull/10152, after the last CI build of this PR.

**Pull request merged**